File: old/src/main/kotlin/logic/StockItemConnection.kt
class StockItemConnection : IStockItemConnection {
    override var buyerName: String
        get() {
            return buyerName
        }

    constructor(buyerName: String) {
        this.buyerName = buyerName;
    }

    override fun messageReceived(): (String) -> Unit {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun sendMessage(message: String) {
        // Send the message through a 3rd party SDK
    }
}

File: new/src/Logic/Connections/StockItemConnection.cs
using System;

namespace AutoBuyer.Logic.Connections
{
    public class StockItemConnection : IStockItemConnection
    {
        public event Action<String> MessageReceived;

        public void SendMessage(String message)
        {
            // Send the message through a 3rd party SDK
        }
    }
}
